### ISSUE TITLE
Doesn't finish decoding stream if schema doesn't have any fields

### ISSUE THREAD

**Issue by HawaiianSpork** [mtth/avsc#198]
If the schema doesn't contain any fields then `createFileEncoder` doesn't read the last object. 

Unit Test:
```javascript
  test('createFileEncoder empty schema', function (cb) {
    var type = types.Type.forSchema({
      type: 'record',
      name: 'Empty',
      fields: []
    });
    var path = tmp.fileSync().name;
    var encoder = index.createFileEncoder(path, type);
    encoder.write({});
    encoder.end({});
    var n = 0;
    encoder.on('finish', function () {
      setTimeout(function () { // Hack to wait until the file is flushed.
        index.createFileDecoder(path)
          .on('data', function (obj) {
            n++;
            assert(type.isValid(obj));
          })
          .on('end', function () {
            assert.equal(n, 2);
            cb();
          });
      }, 50);
    });
  });
```

Unit Test Output:
```
  1) index
       createFileEncoder empty schema:

      Uncaught AssertionError: 1 == 2
      + expected - actual

      -1
      +2
```

It appears this is a problem with decoding rather then encoding because I get the same results when using a different encoder.

**Comment by mtth** [comment]
Interesting find! Empty records have the unusual property of using 0 bytes when encoded, which wasn't compatible with one of the assumptions in the code. https://github.com/mtth/avsc/commit/776d9e65178a152ccb849e07bde7bf8d019152ab fixes this, now released in `5.4.2`.

**Comment by HawaiianSpork** [comment]
Thanks for the quick fix.
